framework/synapse-framework-exception/src/main/java/io/americanexpress/synapse/framework/exception/model/HttpStatus.java

```diff
@@ -0,0 +1,232 @@
+
+    private HttpStatus(int value, Series series, String reasonPhrase) {
+        this.value = value;
+        this.series = series;
+        this.reasonPhrase = reasonPhrase;
+    }
+
+    public int value() {
+        return this.value;
+    }
+
+    public Series series() {
+        return this.series;
+    }
+
+    public String getReasonPhrase() {
+        return this.reasonPhrase;
+    }
+
+    public boolean is1xxInformational() {
+        return this.series() == HttpStatus.Series.INFORMATIONAL;
+    }
+
+    public boolean is2xxSuccessful() {
+        return this.series() == HttpStatus.Series.SUCCESSFUL;
+    }
+
+    public boolean is3xxRedirection() {
+        return this.series() == HttpStatus.Series.REDIRECTION;
+    }
+
+    public boolean is4xxClientError() {
+        return this.series() == HttpStatus.Series.CLIENT_ERROR;
+    }
+
+    public boolean is5xxServerError() {
+        return this.series() == HttpStatus.Series.SERVER_ERROR;
+    }
+
+    public boolean isError() {
+        return this.is4xxClientError() || this.is5xxServerError();
+    }
+
+    public String toString() {
+        int var10000 = this.value;
+        return "" + var10000 + " " + this.name();
+    }
+
+    public static HttpStatus valueOf(int statusCode) {
+        HttpStatus status = resolve(statusCode);
+        if (status == null) {
+            throw new IllegalArgumentException("No matching constant for [" + statusCode + "]");
+        } else {
+            return status;
+        }
+    }
+
+    @Nullable
+    public static HttpStatus resolve(int statusCode) {
+        HttpStatus[] var1 = VALUES;
+        int var2 = var1.length;
+
+        for(int var3 = 0; var3 < var2; ++var3) {
+            HttpStatus status = var1[var3];
+            if (status.value == statusCode) {
+                return status;
+            }
+        }
+
+        return null;
+    }
+
+    public static enum Series {
+        INFORMATIONAL(1),
+        SUCCESSFUL(2),
+        REDIRECTION(3),
+        CLIENT_ERROR(4),
+        SERVER_ERROR(5);
+
+        private final int value;
+
+        private Series(int value) {
+            this.value = value;
+        }
+
+        public int value() {
+            return this.value;
+        }
+
+        /** @deprecated */
+        @Deprecated
+        public static Series valueOf(HttpStatus status) {
+            return status.series;
+        }
+
+        public static Series valueOf(int statusCode) {
+            Series series = resolve(statusCode);
+            if (series == null) {
+                throw new IllegalArgumentException("No matching constant for [" + statusCode + "]");
+            } else {
+                return series;
+            }
+        }
+
+        @Nullable
+        public static Series resolve(int statusCode) {
+            int seriesCode = statusCode / 100;
+            Series[] var2 = values();
+            int var3 = var2.length;
+
+            for(int var4 = 0; var4 < var3; ++var4) {
```

Comment:
same here as above

Reply:
Same, this will go away after exceptions re-work.